src/Input/TextField/TextField.test.tsx

```diff
@@ -0,0 +1,148 @@
+    expect(textFieldInput.value).toEqual('');
+  });
+});
+
+describe('when a value is entered', () => {
+  it('should call onChange once', () => {
+    setupTextFieldWithChangeEvent();
+    expect(props.onChange).toHaveBeenCalledTimes(inputValue.length);
+  });
+
+  it('should return the input value when onChange is passed a function: event => event.target.value', () => {
+    setupTextFieldWithChangeEvent();
+    expect(props.onChange.mock.results.slice(-1)[0].value).toEqual(inputValue);
+  });
+
+  it('should display the correct value', () => {
+    const { textFieldInput } = setupTextFieldWithChangeEvent();
+    expect(textFieldInput.value).toEqual(inputValue);
+  });
+});
+
+describe('when status is:', () => {
+  it("'success', it should display a lightblack border", () => {
+    const { textFieldInput } = setupTextField();
+    expect(textFieldInput).toHaveStyle(`
+      border-color: ${SecondaryColor.lightblack};
+    `);
+  });
+
+  it("'error', it should display a red border", () => {
+    const { textFieldInput } = setupTextField({ status: 'error' });
+    expect(textFieldInput).toHaveStyle(`
+      border-color: ${PrimaryColor.glintsred};
+    `);
+  });
+});
+
+describe('when disabled is true', () => {
+  it('should display cursor: not-allowed', () => {
+    const { textFieldInput } = setupTextField({ disabled: true });
+    expect(textFieldInput).toHaveStyle('cursor: not-allowed;');
+  });
+
+  it('should be disabled', () => {
+    const { textFieldInput } = setupTextField({ disabled: true });
+    expect(textFieldInput).toBeDisabled();
+  });
+});
+
+describe('when small is:', () => {
+  it('true, it should display the correct size', () => {
+    const { textFieldInput } = setupTextField({ small: true });
+    expect(textFieldInput).toHaveStyle('padding: 13px 15px;');
+  });
+
+  it('false, it should display the correct size', () => {
+    const { textFieldInput } = setupTextField({ disabled: true });
+    expect(textFieldInput).toHaveStyle('padding: 15px 20px;');
+  });
+});
+
+describe('when disableTyping is true', () => {
+  it('should display cursor: pointer', () => {
+    const { textFieldInput } = setupTextField({ disableTyping: true });
+    expect(textFieldInput).toHaveStyle('cursor: pointer;');
+  });
+
+  it('should not be able to input a value', () => {
+    const { textFieldInput } = setupTextField({
+      disableTyping: true,
+    });
+    userEvent.type(textFieldInput, inputValue);
+    expect(textFieldInput.value).toBe('');
+  });
+});
+
+describe('when removeFloatingLabel is true', () => {
+  it('should remove the floating label', () => {
+    const { textFieldInput } = setupTextField({ removeFloatingLabel: true });
+    const textFieldLabel = textFieldInput.parentElement.querySelector(
+      '[data-testid="textfield-label"]'
+    ) as HTMLLabelElement;
+    expect(textFieldInput.parentElement).not.toContainElement(textFieldLabel);
+  });
+
+  it('should display a placeholder', () => {
```

Comment:
should we check the placeholder value as well?

Reply:
```
const textFieldInput = utils.getByPlaceholderText(props.label);
```

`getByPlaceholderText` already checks for it. It will throw an error if it cannot find a label with the value `props.label`